src/main/java/software/amazon/cloudformation/LambdaWrapper.java

```diff
@@ -200,12 +211,16 @@ public void handleRequest(final InputStream inputStream, final OutputStream outp
             // deserialize incoming payload to modelled request
             try {
                 request = this.serializer.deserialize(input, typeReference);
+
+                handlerResponse = processInvocation(rawInput, request, context);
             } catch (MismatchedInputException e) {
                 JSONObject resourceSchemaJSONObject = provideResourceSchemaJSONObject();
                 JSONObject rawModelObject = rawInput.getJSONObject("requestData").getJSONObject("resourceProperties");
                 this.validator.validateObject(rawModelObject, resourceSchemaJSONObject);
+                handlerResponse = ProgressEvent.defaultFailureHandler(new TerminalException("#: Invalid input provided", e),
```

Comment:
Why  using "#: " at the beginning of the error message?   Also TerminalException seems not right, how about CfnInvalidRequestException?